app/webserver/helpers.go

```diff
@@ -224,12 +224,50 @@ func updateGroup(updateGroupData UpdateGroupMessage) *store.Session {
 		Members: members,
 	}
 	store.SaveGroup(store.Groups[group.Hexid])
-	session := store.SessionsModel.GetByE164(group.Hexid)
-	msg := session.Add(store.GroupUpdateMsg(updateGroupData.Members, updateGroupData.Name), "", []store.Attachment{}, "", true, store.ActiveSessionID)
-	msg.Flags = helpers.MsgFlagGroupNew
-	//qml.Changed(msg, &msg.Flags)
-	store.SaveMessage(msg)
+	session, err := store.SessionsModel.GetByUUID(group.Hexid)
+	if err != nil {
+		ShowError(err.Error())
+	} else {
+		msg := session.Add(store.GroupUpdateMsg(updateGroupData.Members, updateGroupData.Name), "", []store.Attachment{}, "", true, store.ActiveSessionID)
+		msg.Flags = helpers.MsgFlagGroupUpdate
+		store.SaveMessage(msg)
+
+		return session
+	}
+	return nil
 
+}
+func joinGroup(joinGroupmessage JoinGroupMessage) *store.Session {
+	log.Infoln("[axolotl] joinGroup", joinGroupmessage.ID)
+	group, err := textsecure.JoinGroup(joinGroupmessage.ID)
+	if err != nil {
+		if group != nil {
+			session, _ := store.SessionsModel.GetByUUID(group.Hexid)
+			session.Name = group.DecryptedGroup.Title
+			session.GroupJoinStatus = group.JoinStatus
+			store.UpdateSession(session)
+		}
+		log.Errorln("[axolotl] joinGroup", err)
+		ShowError(err.Error())
+		return nil
+	}
+	log.Infoln("[axolotl] joining group was succesful", group.Hexid)
+	store.Groups[group.Hexid] = &store.GroupRecord{
+		GroupID: group.Hexid,
+		Name:    group.DecryptedGroup.Title,
+		// Members: members,
+		JoinStatus: store.GroupJoinStatusJoined,
+	}
+	store.SaveGroup(store.Groups[group.Hexid])
```

Comment:
I don't understand what you mean, we have the problem, that we have the group and the session and group != session, meaning we have somehow value duplication because axolotl-web is currently using only the session and not the group object.

Reply:
`store.Groups[group.Hexid]` is used like a variable in your code.
I just propose 
```golang
storeGroup := &store.GroupRecord{
		GroupID:    group.Hexid,
		Name:       group.DecryptedGroup.Title,
		Members:    members,
		JoinStatus: store.GroupJoinStatusJoined,
	}
store.Groups[group.Hexid] = storeGroup
store.SaveGroup(storeGroup)
```
instead of line 259..265